primitives/block/src/macro_block.rs

```diff
@@ -81,6 +82,29 @@ pub struct MacroBody {
     pub disabled_set: BitSet,
 }
 
+impl<'a> TryFrom<&'a MacroBlock> for ZKPMacroBlock {
+    type Error = ();
+
+    fn try_from(block: &'a MacroBlock) -> Result<ZKPMacroBlock, Self::Error> {
+        Ok(if let Some(proof) = block.justification.as_ref() {
+            ZKPMacroBlock {
+                block_number: block.block_number(),
+                round_number: proof.round,
+                header_hash: block.hash().into(),
+                signature: proof.sig.signature.get_point(),
+                signer_bitmap: proof
+                    .sig
+                    .signers
+                    .iter_bits()
+                    .take(policy::SLOTS as usize)
+                    .collect(),
+            }
+        } else {
+            ZKPMacroBlock::without_signatures(block.block_number(), 0, block.hash().into())
+        })
```

Comment:
I think I slightly prefer this.
```suggestion
        if let Some(proof) = block.justification.as_ref() {
            Ok(ZKPMacroBlock {
                block_number: block.block_number(),
                round_number: proof.round,
                header_hash: block.hash().into(),
                signature: proof.sig.signature.get_point(),
                signer_bitmap: proof
                    .sig
                    .signers
                    .iter_bits()
                    .take(policy::SLOTS as usize)
                    .collect(),
            })
        } else {
            Ok(ZKPMacroBlock::without_signatures(block.block_number(), 0, block.hash().into()))
        })
```